src/ConfigManager.ts

```diff
@@ -3,18 +3,49 @@ import Chokidar from 'chokidar';
 import YAML from 'yaml';
 import logger from './Logger';
 import ProxyHost from './ProxyHost';
+import EventEmitter from 'events';
 
 export default class ConfigManager {
   private configFile = 'config/config.yml';
   private proxyHosts: Map<string, ProxyHost>;
+  private proxyListeningPort: number | null;
+  private eventEmitter: EventEmitter;
 
   constructor(proxyHosts: Map<string, ProxyHost>) {
     this.proxyHosts = proxyHosts;
+    this.proxyListeningPort = null;
+    this.eventEmitter = new EventEmitter();
     this.createIfNotExist();
     this.parseConfig();
     this.watch();
   }
 
+  private static parsePort(p: string): number | null {
+    const PORT = parseInt(p, 10);
+    if (Number.isInteger(PORT) && PORT >= 0 && PORT <= 49151) {
+      return PORT;
+    }
+
+    return null;
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  public on(event: string, cb: (...args: unknown[]) => void): void {
+    this.eventEmitter.on(event, cb);
+  }
+
+  public getProxyListeningPort(): number {
+    if (this.proxyListeningPort) return this.proxyListeningPort;
+
+    if (process.env.PORT) {
+      this.proxyListeningPort = ConfigManager.parsePort(process.env.PORT);
```

Comment:
Could you please use the env variable `CN_PORT` for this? This way the naming is kept consistent.